src/Behat/Context/MediaContext.php

```diff
@@ -77,7 +77,7 @@ public function iUploadMediaWithNameToField($media_title, $field, $widget) {
     if ($widget == 'media_library') {
       $this->uiContext->iClickOnTheElementWithXpath("//input[contains(@id, 'edit-" . $field . "-open-button')]");
       $this->waitingContext->iWaitForAjaxToFinish(30);
-      $xpath = "//*[@id='drupal-modal']//label[text()='Select " . $media_title . "']/following-sibling::input";
+      $xpath = "//*[@id='drupal-modal']//label[contains(text(),'". $media_title ."')]/following-sibling::input";
```

Comment:
This works in most cases but with this strategy, we are not sure if we are clicking the correct button. There may be more buttons in special cases.

To solve this, can you check if the <input> element that is being selected at the end of the Xpath has a specific attribute that allow us to be sure that is the correct element? If so, please add that attribute to the XPath query

Reply:
Xpath improved. Tested on a project that needed this fix.